src/data/mystery-encounters/encounters/weird-dream-encounter.ts

```diff
@@ -209,7 +222,88 @@ export const WeirdDreamEncounter: MysteryEncounter =
         ],
       },
       async (scene: BattleScene) => {
-        // Reduce party levels by 20%
+        // Battle your "future" team for some item rewards
+        const transformations: PokemonTransformation[] = scene.currentBattle.mysteryEncounter!.misc.teamTransformations;
+
+        // Uses the pokemon that player's party would have transformed into
+        const enemyPokemonConfigs: EnemyPokemonConfig[] = [];
+        for (const transformation of transformations) {
+          const newPokemon = transformation.newPokemon;
+          const previousPokemon = transformation.previousPokemon;
+
+          await postProcessTransformedPokemon(scene, previousPokemon, newPokemon, newPokemon.species.getRootSpeciesId(), true);
+
+          const dataSource = new PokemonData(newPokemon);
+          dataSource.player = false;
+
+          // Copy all held items
+          const heldItems = previousPokemon.getHeldItems();
+          const newPokemonHeldItemConfigs: HeldModifierConfig[] = heldItems.map(previousMod => {
+            return {
+              modifier: previousMod.clone() as PokemonHeldItemModifier,
+              stackCount: previousMod.getStackCount(),
+              isTransferable: false
+            };
+          });
+
+          if (newPokemonHeldItemConfigs.filter(config => config.modifier instanceof PokemonBaseStatFlatModifier).length === 0) {
+            // Also add Old Gateau (even on transformed mons that shouldn't normally get it)
+            newPokemonHeldItemConfigs.push({
+              modifier: generateModifierType(scene, modifierTypes.MYSTERY_ENCOUNTER_OLD_GATEAU) as PokemonHeldItemModifierType,
+              stackCount: 1,
+              isTransferable: false
+            });
```

Comment:
```suggestion
            // Every opponent pokemon (regardless of BST) gets +20 permanent BST to 3 stats:  lowest of HP/Spd, lowest of Atk/SpAtk, and lowest of Def/SpDef
            const stats: Stat[] = [];
            const baseStats = newPokemon.getSpeciesForm().baseStats.slice(0);
            // HP or Speed
            stats.push(baseStats[Stat.HP] < baseStats[Stat.SPD] ? Stat.HP : Stat.SPD);
            // Attack or SpAtk
            stats.push(baseStats[Stat.ATK] < baseStats[Stat.SPATK] ? Stat.ATK : Stat.SPATK);
            // Def or SpDef
            stats.push(baseStats[Stat.DEF] < baseStats[Stat.SPDEF] ? Stat.DEF : Stat.SPDEF);
            newPokemonHeldItemConfigs.push({
              modifier: generateModifierType(scene, modifierTypes.MYSTERY_ENCOUNTER_OLD_GATEAU, [20, stats]) as PokemonHeldItemModifierType,
              stackCount: 1,
              isTransferable: false
            });
```
To ensure that Old Gateau also generates properly for the "future self" opponent, giving +20 to stats
(This is basically copied from the code for generating Old Gateau for the player's transformed Pokemon)

Reply:
Ah, good point. I think the better way would be to remove that code lines 249-256, and instead give the old Gateau to all Pokemon if they are for generated for battle in `postProcessTransformedPokemon`, by replacing this:

https://github.com/pagefaultgames/pokerogue/blob/751e871b022b0c38a7c8c03d9844383b13c3e7d9/src/data/mystery-encounters/encounters/weird-dream-encounter.ts#L523

with: 
`if (forBattle || newPokemon.getSpeciesForm().getBaseStatTotal() < NON_LEGENDARY_BST_THRESHOLD) { `

Does that sound right to you?

Reply:
Maybe that should be a question for balance though.
If we leave it like it is right now this they get a random stat increase from 0-20 to Stat.HP, Stat.ATK, Stat.DEF
If we change to make it generate the same way that makes the fight way tougher